src/nvim/mark.c

```diff
@@ -1204,16 +1214,27 @@ void cleanup_jumplist(win_T *wp, bool checktail)
         break;
       }
     }
-    if (i >= wp->w_jumplistlen) {  // no duplicate
+    bool mustfree = false;
```

Comment:
this is a "dead assignment", static analysis will flag it.

```suggestion
    bool mustfree;
```

I would probably lean towards a ternary or some other way to be less verbose in the logic below, but I suppose that could turn out worse.

Reply:
Fixed the dead store in https://github.com/neovim/neovim/pull/11530/commits/7f294976248812aa102e1eb947b5e3f1694852c3 .  The less verbose versions I've been able to come up with so far have been less readable (highly subjective), but if there's a better way to write this let's do it.